src/parser.jison

```diff
@@ -44,7 +44,7 @@
 (([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)dpcm\b            return 'RES';
 (([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)dppx\b            return 'RES';
 (([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)\%                return 'PERCENTAGE';
-(([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)\b                return 'NUMBER';
+(([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)(?!".")\b         return 'NUMBER';
 
 (([0-9]+("."[0-9]+)?|"."[0-9]+)(e(\+|-)[0-9]+)?)-?([a-zA-Z_]|[\240-\377]|(\\[0-9a-fA-F]{1,6}(\r\n|[ \t\r\n\f])?|\\[^\r\n\f0-9a-fA-F]))([a-zA-Z0-9_-]|[\240-\377]|(\\[0-9a-fA-F]{1,6}(\r\n|[ \t\r\n\f])?|\\[^\r\n\f0-9a-fA-F]))*\b       return 'UNKNOWN_DIMENSION';
```

Comment:
Alternatively we could try simplifying `UNKNOWN_DIMENSION`. Why does it need to match so many types of characters, even `\r, \t`, etc? According to the spec, the unit can only be a CSS identifier https://www.w3.org/TR/css-values-3/#css-identifier

Reply:
I think it is right direction, yes, I think it is wrong

Reply:
I realized that this parsing duplicates a lot of work that's already done in postcss-value-parser. It [calls postcss-value-parser, stringifies the result, then parses again](https://github.com/postcss/postcss-calc/blob/1f2513d79aa442d25c58bf4577b5e59bf5b65c77/src/lib/transform.js#L27). Looks like the reason is that this package also wants to parse `calc()` in selectors, which is not in the spec. But it feels a bit like a waste to do this work all over again. There's already 3 separate CSS tokenizers in postcss, postcss-value-parser, and postcss-selector-parser. So together with postcss-calc, the tokenizer has been reimplemented 4 times in PostCSS projects.

Reply:
Yes, It was here before I was invited and fix bugs, will be great to rewrite it on one parser